### ISSUE TITLE
inject: `--linkerd-version` override is not applied

### ISSUE THREAD

**Issue by olix0r** [linkerd/linkerd2#2551]
```
linkerd inject strest.yml --linkerd-version=ver-reconnect-backoff  2>/dev/null |grep -c ver-reconn
0
```

**Comment by ihcsim** [comment]
Related to #2497.

**Comment by olix0r** [comment]
It's convenient to fix this with #2556, so i'll take it

**Comment by ihcsim** [comment]
Fixed under #2547.

**Comment by ihcsim** [comment]
The overridden linkerd version needs to be encoded as annotation. Re-opening issue.
